[8.6.0] - Unable to Save custom module record through Subpanel interface

#### Issue
An "Error occurred while saving record" error is returned whenever you attempt to save a record whose creation was initiated by clicking on the **Action -> Create** button within the module's Subpanel on a parent record's page. This only occurs when trying to create records for a custom module, and only when trying to create a new record through a Subpanel.

This error does not occur on our Production server running 8.5.0.

#### Expected Behavior
You should be able to navigate to a Subpanel for a record, click the Create button to create a new record, and save the record.

#### Actual Behavior
Creating a record for a custom module from a Subpanel fails when attempting to save that record. 

suitecrm.log reveals that the error is caused by a field validation error for the 'id' key.

`Sun Apr 21 21:24:06 2024 [1408][2e14df70-84db-312f-4c10-65c10f69d2b3][ERROR] ApiBeanMapper field validation | Key 'id' | Error 'Invalid id field 'ssvp_households_ssvp_membersssvp_households_ida'. Value not a string nor a number'`

ApiBeanMapper is expecting a string or number, but when inspecting the payload of the POST request it appears it is trying to pass an array of objects for that ID field:

`{
  "input": {
    "module": "ssvp_Members",
    "attributes": {
      "assigned_user_id": "2e14df70-84db-312f-4c10-65c10f69d2b3",
      "assigned_user_name": {
        "id": "2e14df70-84db-312f-4c10-65c10f69d2b3",
        "user_name": "admin"
      },
      "relate_to": "ssvp_households_ssvp_members",
      "relate_id": "93cc0bf6-3d2a-5a53-e929-65bc42eaf0c6",
      "ssvp_households_ssvp_members_name": {
        "name": "Smith, John",
        "id": "93cc0bf6-3d2a-5a53-e929-65bc42eaf0c6"
      },
      "ssvp_households_ssvp_membersssvp_households_ida": [
        {
          "id": "93cc0bf6-3d2a-5a53-e929-65bc42eaf0c6",
          "module": "ssvp_Households",
          "attributes": {
            "id": "93cc0bf6-3d2a-5a53-e929-65bc42eaf0c6",
            "name": "Smith, John"
          }
        }
      ],
      "ssvp_households_ssvp_members": [
        {
          "id": "93cc0bf6-3d2a-5a53-e929-65bc42eaf0c6",
          "module": "ssvp_Households",
          "attributes": {
            "id": "93cc0bf6-3d2a-5a53-e929-65bc42eaf0c6",
            "name": "Smith, John"
          }
        }
      ],
    }
  }
}`


#### Possible Fix
<!--- Not obligatory, but suggest a fix or reason for the bug -->

#### Steps to Reproduce
1. Navigate to a record for a module that has a One-to-Many relationship with a custom module.
2. Expand the Subpanel for the custom module and click **Action -> Create**
3. Fill out the form for the record and click **Save**
4. "Error occurred while saving record" error will appear and you will be redirected to the parent record.

#### Context
This bug disrupts our workflow as we utilize a number of custom modules with relationships to other modules. Currently, we are able to get around this issue by creating the record normally outside of the Subpanel. 

Because of this workaround, I would rate this as a low/medium priority.

#### Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* SuiteCRM Version used: 8.6.0
* Browser name and version: Google Chrome Version 123.0.6312.123 (Official Build) (64-bit)
* Environment name and version (e.g. MySQL, PHP 7): MySQL 8.0.36, PHP 8.1
* Operating System and version (e.g Ubuntu 16.04): Ubuntu 22.04

--- a/public/legacy/include/portability/Bean/Field/Validation/IdFieldValidator.php
+++ b/public/legacy/include/portability/Bean/Field/Validation/IdFieldValidator.php
@@ -15,6 +15,10 @@ class IdFieldValidator implements FieldValidatorInterface
             return '';
         }
 
+        if (is_array($value)) {
+            $value = $value[0]['id'] ?? $value['id'];
+        }
+
         if (!is_string($value) && !is_numeric($value)) {
             return "Invalid id field '$field'. Value not a string nor a number";
         }

--- a/public/legacy/modules/ModuleBuilder/parsers/relationships/AbstractRelationship.php
+++ b/public/legacy/modules/ModuleBuilder/parsers/relationships/AbstractRelationship.php
@@ -453,14 +453,16 @@ class AbstractRelationship
                     break ;
                 default:
                     // now see if we have any module inheriting from the 'file' template - records in file-type modules are named by the document_name field, not the usual 'name' field
-                    $object = $GLOBALS ['beanList'] [ $sourceModule ];
-                    require_once($GLOBALS ['beanFiles'] [ $object ]);
-                    $bean = new $object();
-                    if (isset($GLOBALS [ 'dictionary' ] [ $object ] [ 'templates'])) {
-                        if (in_array('file', $GLOBALS [ 'dictionary' ] [ $object ] [ 'templates'])) {
-                            $vardef [ 'rname' ] = 'document_name' ;
-                        } elseif (in_array('person', $GLOBALS [ 'dictionary' ] [ $object ] [ 'templates'])) {
-                            $vardef [ 'db_concat_fields' ] = array( 0 =>'first_name', 1 =>'last_name') ;
+                    $object = $GLOBALS ['beanList'] [ $sourceModule ] ?? '';
+                    if (!empty($object)) {
+                        require_once($GLOBALS ['beanFiles'] [$object]);
+                        $bean = new $object();
+                        if (isset($GLOBALS ['dictionary'] [$object] ['templates'])) {
+                            if (in_array('file', $GLOBALS ['dictionary'] [$object] ['templates'])) {
+                                $vardef ['rname'] = 'document_name';
+                            } elseif (in_array('person', $GLOBALS ['dictionary'] [$object] ['templates'])) {
+                                $vardef ['db_concat_fields'] = array(0 => 'first_name', 1 => 'last_name');
+                            }
                         }
                     }
 

